core/templates/dev/head/domain/learner_dashboard/LearnerDashboardIconsDirective.js

```diff
@@ -0,0 +1,94 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for showing learner dashboard icons.
+ */
+
+oppia.directive('learnerDashboardIcons', [
+  'UrlInterpolationService', function(UrlInterpolationService) {
+    return {
+      restrict: 'E',
+      scope: {
+        getActivityType: '&activityType',
+        getActivityId: '&activityId',
+        getActivityTitle: '&activityTitle',
+        activityActive: '=activityActive'
+      },
+      templateUrl:  UrlInterpolationService.getDirectiveTemplateUrl(
+        '/domain/learner_dashboard/' +
+        'learner_dashboard_icons_directive.html'),
+      controller: [
+        '$scope', 'LearnerDashboardIdsBackendApiService',
+        'LearnerDashboardActivityIdsObjectFactory',
+        'LearnerPlaylistService',
+        function($scope, LearnerDashboardIdsBackendApiService,
+                 LearnerDashboardActivityIdsObjectFactory,
+                 LearnerPlaylistService) {
+          $scope.activityIsActive = true;
+
+          $scope.$watch('activityActive', function(value) {
+            $scope.activityIsActive = !$scope.activityIsActive;
+          });
+
+          LearnerDashboardIdsBackendApiService.fetchLearnerDashboardIds().then(
+            function(response) {
+              $scope.learnerDashboardActivityIds = (
+                LearnerDashboardActivityIdsObjectFactory.createFromBackendDict(
+                  response.data.learner_dashboard_activity_ids));
+            }
+          );
+
+          $scope.toggleActivityIsActive = function(activityId) {
+            $scope.activityIsActive = !$scope.activityIsActive;
+          };
+
+          $scope.canActivityBeAddedToLearnerPlaylist = function(
+            activityId) {
+            if ($scope.learnerDashboardActivityIds) {
+              /* eslint-disable max-len */
+              if ($scope.learnerDashboardActivityIds.belongsToLearnerDashboardActivities(
+                activityId)) {
+              /* eslint-enable max-len */
+                return false;
+              } else {
+                return $scope.activityIsActive;
+              }
+            }
+          };
+
+          $scope.addToLearnerPlaylist = function(activityId, activityType) {
+            var isSuccessfullyAdded = (
+              LearnerPlaylistService.addToLearnerPlaylist(
+                activityId, activityType));
+            if (isSuccessfullyAdded) {
+              /* eslint-disable max-len */
+              $scope.learnerDashboardActivityIds.addToExplorationLearnerPlaylist(
```

Comment:
I don't understand why we have "exploration" in the name, when there's a general activity type here...

Reply:
Fixed!